x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/session/EsqlResolvedIndexExpressionIT.java

```diff
@@ -55,14 +54,27 @@ public void testLocalAndRemote() {
 
         assertThat(
             resolveIndices("index-*,*:index-*"),
-            allOf(
-                hasEntry(LOCAL_CLUSTER, resolvedIndexExpression("index-*", "index-1")),
-                hasEntry(REMOTE_CLUSTER_1, resolvedIndexExpression("index-*", "index-2")),
-                hasEntry(REMOTE_CLUSTER_2, resolvedIndexExpression("index-*", "index-3"))
+            equalTo(
+                Map.ofEntries(
+                    Map.entry(LOCAL_CLUSTER, resolvedIndexExpression("index-*", "index-1")),
+                    Map.entry(REMOTE_CLUSTER_1, resolvedIndexExpression("index-*", "index-2")),
+                    Map.entry(REMOTE_CLUSTER_2, resolvedIndexExpression("index-*", "index-3"))
+                )
             )
         );
     }
 
+    public void testRemoteIndices() {
+        createIndex(LOCAL_CLUSTER, "index-1");
+        createIndex(REMOTE_CLUSTER_1, "index-2");
+        createIndex(REMOTE_CLUSTER_2, "index-3");
+
+        assertThat(
+            resolveIndices(REMOTE_CLUSTER_1 + ":index-*"),
+            equalTo(Map.of(REMOTE_CLUSTER_1, resolvedIndexExpression("index-*", "index-2")))
```

Comment:
This was previously incorrectly resolving to:
```
Map.of(
  LOCAL_CLUSTER, resolvedIndexExpression("", "")
  REMOTE_CLUSTER_1, resolvedIndexExpression("index-*", "index-2")
)
```
I have added a check to remove empty entries.